src/mpi.jl

```diff
@@ -0,0 +1,10 @@
+# These are just dummy functions. We only implement real
+# versions if MPI.jl is loaded to avoid letting TrixiBase.jl
+# depend explicitly on MPI.jl.
+# We dispatch on the return type of get_extension to provide a fallback
+
+mpi_isparallel() = mpi_isparallel_internal(Base.get_extension(TrixiBase, :TrixiBaseMPIExt))
```

Comment:
`Base.get_extension` is not a free operation, and my expectation for a query like this is that should be fast.
Moreover, we add a dynamic dispatch.